.github/workflows/ci-pipeline.yml

```diff
@@ -18,3 +18,42 @@ jobs:
     - run: docker build --target test --tag test .
     - run: docker run test todo_app/test/unit_tests
     - run: docker run --env-file .env.docker test todo_app/test/integration_tests
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ secrets.DOCKERHUB_USERNAME }}/dev-ops-todo
+      
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: true
+          target: production
+          tags: dsdanielh/dev-ops-todo:prod2
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
I've never seen these labels used before - interesting!

It's interesting to discuss how you're achieving that by just duplicating the current labels rather than specifying them explicitly in the pipeline - that's losing some advantages of the general infra-as-code by having those clearly documented & consistent, but it is a bit more flexible to start with